### PR TITLE
Trigger sto scan to block pull requests with vulnerabilities sto 7342

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
Preview link: [Trigger STO scans to block pull requests](https://660ff64f62a91613717e214e--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities)

## Merge requirements

- [x] Successful preview build.
- [x] No merge conflicts.
- [x] Up to date with main.
- [ ] SME/Product review.
- [x] Editorial/peer review.
- [x] No open change requests from reviewers.

### CODE REVIEW

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+title: Trigger STO scans to block pull requests with vulnerabilities
+description: How to trigger STO scans to block PRs with vulnerabilities.
+sidebar_label: Trigger STO scans to block pull requests
+sidebar_position: 5
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+You can create Git event triggers to support a variety of STO workflows and use case. This topic describes how to do the following:
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:11]:
```suggestion
You can create Git event triggers to support a variety of STO workflows and use cases. This topic describes how to do the following: 
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+2. [Create a pull request](#test-the-outbound-webhook-and-trigger) to test the webhook and trigger. 
+
+3. [Add a branch protection rule](#add-a-branch-protection-rule) to to ensure that the pull request cannot be merged if the Harness pipeline fails.
+
+
+## Before you begin
+
+These workflows require the following:
+
+- A [**connector**](/docs/category/code-repo-connectors/) to your Git service
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:33]:
```suggestion
- A [**connector**](/docs/category/code-repo-connectors/) to your Git service.
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+
+<summary>Trigger setup</summary>
+
+1. Go to your STO pipeline, select **Triggers** (top right), and add a new trigger.
+
+2. Set up the trigger as follows. 
+
+#### [Configuration](/docs/platform/triggers/triggering-pipelines#configure-the-trigger)
+
+1. [**Connector**](/docs/category/code-repo-connectors/) to your Git service
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:66]:
```suggestion
1. [**Connector**](/docs/category/code-repo-connectors/) to your Git service.
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+
+2. **Repository name**
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:68]:
```suggestion
2. **Repository name**.
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+
+3. [**Event**](/docs/platform/triggers/triggers-reference#event-and-actions) = **Pull Request**
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:70]:
```suggestion
3. [**Event**](/docs/platform/triggers/triggers-reference#event-and-actions) = **Pull Request**.
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+   Here's a simple example: trigger a build if a PR seeks to update a specific `pom.xml` in the `main` branch. 
+
+   <DocImage path={require('./static/trigger-to-block-prs/changed-file-condition-example-simple.png')} width="50%" height="50%" title="Add shared path for scan results" /> 
+
+<br/>
+<br/>
+
+#### [Pipeline input](/docs/platform/triggers/triggering-pipelines#set-pipeline-input) 
+   
+   The pipeline input should be preconfigured correctly, with  **Build Type** set to **Git Pull Request**.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:91]:
```suggestion
   The pipeline input should be preconfigured correctly, with **Build Type** set to **Git Pull Request**.
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+
+#### Test the trigger
+
+After you create the trigger, proceed to [Test the outbound webhook and trigger](#test-the-outbound-webhook-and-trigger)
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:95]:
```suggestion
After you create the trigger, proceed to [Test the outbound webhook and trigger](#test-the-outbound-webhook-and-trigger).
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+
+</details>
+
+
+
+### Trigger on a PR review comment
+
+You can specify a trigger that says: If a reviewer includes a specific keyword in a pull-request review comment, run the pipeline and scan the repo.
+
+This type of trigger is useful when a pipeline execution fails for reasons other than [Fail on Severity](/docs/security-testing-orchestration/get-started/key-concepts/fail-pipelines-by-severity/). If the STO scan doesn't finish in the  original execution, a reviewer can add a review comment with a keyword such as `RERUN_STO_PIPELINE`.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:162]:
```suggestion
This type of trigger is useful when a pipeline execution fails for reasons other than [Fail on Severity](/docs/security-testing-orchestration/get-started/key-concepts/fail-pipelines-by-severity/). If the STO scan doesn't finish in the original execution, a reviewer can add a review comment with a keyword such as `RERUN_STO_PIPELINE`. 
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 9]
```diff
@@ -0,0 +1,368 @@
...
+
+<summary>Trigger setup</summary>
+
+1. Go to your STO pipeline, select **Triggers** (top right), and add a new trigger.
+
+2. Set up the trigger as follows. 
+
+#### [Configuration](/docs/platform/triggers/triggering-pipelines#configure-the-trigger)
+
+1. [**Connector**](/docs/category/code-repo-connectors/) to your Git service
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:174]:
```suggestion
1. [**Connector**](/docs/category/code-repo-connectors/) to your Git service.
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 10]
```diff
@@ -0,0 +1,368 @@
...
+
+2. **Repository name**
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:176]:
```suggestion
2. **Repository name**.
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 11]
```diff
@@ -0,0 +1,368 @@
...
+
+3. [**Event**](/docs/platform/triggers/triggers-reference#event-and-actions) = **Issue Comment**
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:178]:
```suggestion
3. [**Event**](/docs/platform/triggers/triggers-reference#event-and-actions) = **Issue Comment**.
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 12]
```diff
@@ -0,0 +1,368 @@
...
+
+      `<+trigger.payload.comment.body>.contains("RERUN_STO_SCAN")`
+
+      <DocImage path={require('./static/trigger-to-block-prs/review-comment-condition-example-simple.png')} width="50%" height="50%" title="Add shared path for scan results" /> 
+
+<br/>
+
+#### [Pipeline input](/docs/platform/triggers/triggering-pipelines#set-pipeline-input) 
+   
+The pipeline input should be preconfigured correctly, with  **Build Type** should be set to **Git Pull Request**.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:198]:
```suggestion
The pipeline input should be preconfigured correctly, with **Build Type** should be set to **Git Pull Request**.
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 13]
```diff
@@ -0,0 +1,368 @@
...
+  
+<br/>
+
+#### Test the trigger
+
+After you create the trigger, proceed to [Test the outbound webhook and trigger](#test-the-outbound-webhook-and-trigger)
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:204]:
```suggestion
After you create the trigger, proceed to [Test the outbound webhook and trigger](#test-the-outbound-webhook-and-trigger).
```

File: docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md [hunk 14]
```diff
@@ -0,0 +1,368 @@
...
+### Test the protection rule
+
+Now that you've set up the rule, trigger another pipeline execution to verify that the rule stops the pull request.
+
+:::note
+
+To verify the branch protection rule, you must ensure that your STO pipeline fails. To configure your pipeline to fail temporarily, you can do one of the following:
+
+- Set [Fail on Severity](/docs/security-testing-orchestration/get-started/key-concepts/fail-pipelines-by-severity/) to **Low** in the scan step of your pipeline. Then scan a repo with known vulnerabilities. 
+- Add a temporary Run step to your pipeline with the command `exit(1)`
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/stop-builds-based-on-scan-results/trigger-sto-scans-to-block-prs-with-vulnerabilities.md:342]:
```suggestion
- Add a temporary Run step to your pipeline with the command `exit(1)`.
```

**Review by brian-f-harness — DISMISSED** [review]
LGTM
Left a couple of minor suggestions to consider. Please ignore if any suggested punctuation is incorrect.

**Review by brian-f-harness — APPROVED** [review]

**Review by pranay-harness — APPROVED** [review]

**Pull request merged**